### PR TITLE
fix: support Qwen3.5 family for DSpark

### PR DESCRIPTION
**Pull request by Dogacel**
## Purpose

Fix DSpark support for Qwen3.5 family models.

Follow up on #46995 by @benchislett .

Part of the KV cache fix is inherited from #45207.

## Test Plan

We have an internal DSpark checkpoint trained using [TorchSpec](https://github.com/lightseekorg/TorchSpec) that we directly load and run our custom benchmark against. Unfortunately I can't share the model.

## Test Result

Before the model only got ~1.3 acceptance on Qwen3.6 35B, considering the bonus token it is near random. Also the `lm_head` weights were empty when inspected. After applying the fixes it got around ~4.5 for block size 8.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by jamesunnc** [comment]
Just wondering if vllm 0.24.0 already support dspark as a new method on speculative decoding?

File: vllm/v1/worker/gpu/spec_decode/dspark/utils.py [hunk 1]
```diff
@@ -52,7 +52,11 @@ def load_dspark_model(target_model: nn.Module, vllm_config: VllmConfig) -> nn.Mo
             del draft_inner.embed_tokens
         draft_inner.embed_tokens = target_embed
 
-    target_lm_head = getattr(target_model, "lm_head", None)
+    # lm_head lives on the language model, not the (possibly multimodal) outer
+    # wrapper (e.g. Qwen3.5-MoE), whose top-level lm_head is None.
+    target_lm_head = getattr(target_language_model, "lm_head", None)
```
Comment (benchislett) [inline vllm/v1/worker/gpu/spec_decode/dspark/utils.py:57]: can you also update DFlash, and any other speculators that might need this?
https://github.com/vllm-project/vllm/blob/main/vllm/v1/worker/gpu/spec_decode/dflash/utils.py#L60

Comment (benchislett) [reply vllm/v1/worker/gpu/spec_decode/dspark/utils.py:57]: maybe `get_target_lm_head` should be a helper?

**Comment by benchislett** [comment]
I'm not quite sure how I feel about the KV cache fix. Could you run some correctness evals on Qwen3.5 + DSpark (e.g. GPQA-D) to ensure that we're not losing any accuracy?

**Comment by mgoin** [comment]
Hey @Dogacel  I saw your PR here when I was working on https://github.com/vllm-project/vllm/pull/47914 for Qwen3.5 DFlash support for zlab’s checkpoints.
Since we’ve landed mine, https://github.com/vllm-project/vllm/pull/45207, and https://github.com/vllm-project/vllm/pull/47698 - I think all you need to do now in this PR is use the `get_target_lm_head` function in DSpark
